### PR TITLE
Integrate v5 net interface

### PR DESCRIPTION
**Pull request by bureddy**
nccl headers : 2.12.12

### CODE REVIEW

File: src/p2p_plugin.c [hunk 1]
```diff
@@ -235,14 +244,13 @@ ncclResult_t nccl_p2p_ib_init(int *num_devs, nccl_ib_dev_t *ncclIbDevs, char *nc
             ncclIbDevs[ncclNIbDevs].maxQp = ncclParamSharpMaxComms();
             ncclNSharpDevs++;
           }
+          pthread_create(ncclIbAsyncThread, NULL, ncclIbAsyncThreadMain, context);
+          pthread_detach(*ncclIbAsyncThread); // will not be pthread_join()'d
+          ncclSetThreadName(*ncclIbAsyncThread, "NCCL IbAsync %2d", ncclNIbDevs);
```
Comment (AddyLaddy) [inline src/p2p_plugin.c:249]: We recently fixed a SEGV bug with ncclSetThreadName where a thread could exit after the detach but before SetThreadName had completed. See commit d5f6a5f210e980eb06ada5733635baa18ada1f22

File: include/timer.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/*************************************************************************
```
Comment (AddyLaddy) [inline include/timer.h:1]: Is this file used in the plugin?

File: src/p2p_plugin.c (hunk 1 above)
Comment (bureddy) [reply src/p2p_plugin.c:249]: is this fix in public NCCL github? I have taken changes from public github

File: include/timer.h (hunk 2 above)
Comment (bureddy) [reply include/timer.h:1]: Yes. it is used in net_ib.cc.  so i have taken same changes

File: src/p2p_plugin.c (hunk 1 above)
Comment (bureddy) [reply src/p2p_plugin.c:249]: i will take those changes when it is public

File: include/p2p_plugin.h [hunk 3]
```diff
@@ -19,45 +22,77 @@
...
   struct   ibv_context* context;
+  int      pdRefs;
+  struct ibv_pd*  pd;
   struct   ncclIbVerbs verbs;
   char     devName[MAXNAMESIZE];
   char     *pciPath;
   int      realPort;
   int      maxQp;
-} nccl_ib_dev_t;
+  struct   ncclIbMrCache mrCache;
```
[the remark this answers is not in the document]
Comment (bureddy) [reply include/p2p_plugin.h:94]: should be same as public github code. am i missing something?
https://github.com/Mellanox/nccl-rdma-sharp-plugins/pull/83/files#diff-70b9000dbde49d38b63ecd9347e5a4004cd7aeacd13daacff6ec09af2145d1eeR600

**Pull request merged**